File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        List<int[]> arr = new ArrayList<>();
        for (int i = 0; i < nums.length; i++) {
            arr.add(new int[]{nums[i], i});
        }
        arr.sort(Comparator.comparing(x -> x[0]));
        int low = 0;
        int high = nums.length - 1;
        while (low < high) {
            int l = arr.get(low)[0];
            int h = arr.get(high)[0];
            if (l + h == target) {
                return new int[]{arr.get(low)[1], arr.get(high)[1]};
            } else if (l + h < target) {
                low++;
            } else {
                high--;
            }
        }
        return new int[]{};
    }
}

File: 0001-two-sum/0001-two-sum.swift
class Solution {
    func twoSum(_ nums: [Int], _ target: Int) -> [Int] {
    var answer: [Int] = []
    var numsWithIndex: [[Int]] = []
    for i in 0..<nums.count {
        numsWithIndex.append([i, nums[i]])
    }
    let sorted: [[Int]] = numsWithIndex.sorted {$0[1] <= $1[1]}
    var low: Int = 0
    var high: Int = nums.count - 1
    while low < high {
        let sum = sorted[low][1] + sorted[high][1]
        if sum == target {
            answer.append(sorted[low][0])
            answer.append(sorted[high][0])
            return answer
        } else if sum > target {
            high -= 1
        } else {
            low += 1
        }
    }
    return answer
}
}